### PR TITLE
[Launch Handler article] Update to latest explainer version

### PR DESCRIPTION
**Pull request by tomayac**
Changes proposed in this pull request:

- Update to latest explainer version.

### CODE REVIEW

**Comment by tomayac** [comment]
@alancutter FYI.

File: site/en/docs/web-platform/launch-handler/index.md [hunk 1]
```diff
@@ -2,16 +2,20 @@
...
 authors:
   - thomassteiner
-updated: 2022-04-22
 ---
 
-The Launch Handler API lets you control how your app is launched, for example, whether it uses an existing or a new window and whether the chosen window is navigated to the launch URL. This also enqueues a `LaunchParams` object in the launched page's `window.launchQueue`, similar to the File Handling API.
+The Launch Handler API lets you control how your app is launched, for example, whether it uses an
+existing or a new window and whether the chosen window is navigated to the launch URL. This also
+enqueues a `LaunchParams` object in the launched page's `window.launchQueue`, similar to the File
+Handling API.
```
Comment (jpmedley) [inline site/en/docs/web-platform/launch-handler/index.md:18]:
```suggestion
existing or a new window and whether the chosen window is navigated to the launch URL. As with
the File Handing API, this also enqueues a `LaunchParams` object in the launched page's `window.launchQueue` Handling API.
```

File: site/en/docs/web-platform/launch-handler/index.md [hunk 2]
```diff
@@ -27,80 +31,69 @@ The Launch Handler API lets you control how your app is launched, for example, w
...
 #### Local testing
 
-To experiment with the Launch Handler API locally, without an origin trial token, enable the `#enable-desktop-pwas-launch-handler` flag in `about://flags`.
+To experiment with the Launch Handler API locally, without an origin trial token, enable the
+`#enable-desktop-pwas-launch-handler` flag in `about://flags`.
 
 #### Register for the origin trial
 
-Starting in Chromium 98, the Launch Handler API will be available as an [origin trial](/docs/web-platform/origin-trials/) in Chromium. The origin trial is expected to end in Chromium 102 (June 15, 2022). [Register here](https://developers.chrome.com/origintrials/#/trials/active).
+Starting in Chromium 98, the Launch Handler API will be available as an
```
Comment (jpmedley) [inline site/en/docs/web-platform/launch-handler/index.md:45]:
```suggestion
Starting in Chromium 98, the Launch Handler API is available as an
```

File: site/en/docs/web-platform/launch-handler/index.md [hunk 3]
```diff
@@ -27,80 +31,69 @@ The Launch Handler API lets you control how your app is launched, for example, w
...
+[origin trial](/docs/web-platform/origin-trials/) in Chromium. The origin trial is expected to end
+in Chromium 106 (October 26, 2022).
```
Comment (jpmedley) [inline site/en/docs/web-platform/launch-handler/index.md:47]:
```suggestion
in Chromium 106 (October 25, 2022).
```

Comment (jpmedley) [reply site/en/docs/web-platform/launch-handler/index.md:47]: This is the date [on the dashboard](https://developer.chrome.com/origintrials/#/view_trial/2978005253598740481).

File: site/en/docs/web-platform/launch-handler/index.md [hunk 4]
```diff
@@ -27,80 +31,69 @@ The Launch Handler API lets you control how your app is launched, for example, w
...
-The Launch Handler API defines two new interfaces. 
-
-`launchParams`
-: An object containing the targetURL to be handled by the consumer. 
-`launchQueue`
-: Queues launches until they are handled by the specified consumer. 
+The Launch Handler API defines two new interfaces.
 
+`launchParams` : An object containing the targetURL to be handled by the consumer. `launchQueue` :
+Queues launches until they are handled by the specified consumer.
```
Comment (jpmedley) [inline site/en/docs/web-platform/launch-handler/index.md:57]:
```suggestion
`LaunchParams` : An object containing the `targetURL` to be handled by the consumer.
`LaunchQueue` : Queues launches until they are handled by the specified consumer.
```

File: site/en/docs/web-platform/launch-handler/index.md [hunk 5]
```diff
@@ -27,80 +31,69 @@ The Launch Handler API lets you control how your app is launched, for example, w
...
 ## The `launch_handler` manifest member
 
 To declaratively specify the launch behavior of your app, add the `launch_handler` manifest member
-to your manifest. It has two sub-fields, `route_to` and `navigate_existing_client`. The former lets
-you control whether a new or an existing client should be launched, and the latter how and if this
-client should be navigated. The Web App Manifest excerpt below shows a file with
-exemplary values that would always route all launches to a new client.
+to your manifest. It has one sub-field called `client_mode`. It lets you control whether a new or an
+existing client should be launched and if this client should be navigated. The Web App Manifest
+excerpt below shows a file with exemplary values that would always route all launches to a new
```
Comment (jpmedley) [inline site/en/docs/web-platform/launch-handler/index.md:64]:
```suggestion
existing client should be launched and if this client should be navigated. The example
below shows a file with exemplary values that would always route all launches to a new
```

File: site/en/docs/web-platform/launch-handler/index.md [hunk 6]
```diff
@@ -27,80 +31,69 @@ The Launch Handler API lets you control how your app is launched, for example, w
...
+    "client_mode": "navigate-new"
   }
 }
 ```
 
-If unspecified, `launch_handler` defaults to
-`{"route_to": "auto", "navigate_existing_client": "always"}`. The allowed values for the sub-fields
-are as follows:
+If unspecified, `launch_handler` defaults to `{"client_mode": "auto"}`. The allowed values for the
+sub-fields are as follows:
```
Comment (jpmedley) [inline site/en/docs/web-platform/launch-handler/index.md:76]:
```suggestion
sub-fields are:
```

File: site/en/docs/web-platform/launch-handler/index.md [hunk 7]
```diff
@@ -114,7 +107,8 @@ if ('launchQueue' in window && 'targetURL' in LaunchParams.prototype) {
 
 ### Using window.launchQueue
 
-In the following code, the function `extractSongID()` is used to extract a `songID` from the URL passed on launch. This is then used to play a song in a music player PWA.
+In the following code, the function `extractSongID()` is used to extract a `songID` from the URL
```
Comment (jpmedley) [inline site/en/docs/web-platform/launch-handler/index.md:110]:
```suggestion
In the following code, the function `extractSongID()` extracts a `songID` from the URL
```

File: site/en/docs/web-platform/launch-handler/index.md [hunk 8]
```diff
@@ -114,7 +107,8 @@ if ('launchQueue' in window && 'targetURL' in LaunchParams.prototype) {
...
+passed on launch. This is then used to play a song in a music player PWA.
```
Comment (jpmedley) [inline site/en/docs/web-platform/launch-handler/index.md:111]:
```suggestion
passed on launch. This is used to play a song in a music player PWA.
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

**Review by jpmedley — APPROVED** [review]

File: site/en/docs/web-platform/launch-handler/index.md [hunk 9]
```diff
@@ -27,80 +30,67 @@ The Launch Handler API lets you control how your app is launched, for example, w
...
-backwards compatibility with existing implementations.
+The `client_mode` property also accepts a list (array) of values, where the first valid value will be
+used. This is to allow new values to be added to the spec without breaking backwards compatibility
+with existing implementations.
 
 For example, if the hypothetical value `"matching-url-client"` were added, sites would specify
-`"route_to": ["matching-url-client", "existing-client"]` to continue to control the behavior of
-older browsers that did not support `"matching-url-client"`.
+`"client_mode": ["matching-url-client", "navigate-existing"]` to continue to control the
+behavior of older browsers that did not support `"matching-url-client"`.
```
Comment (alancutter) [inline site/en/docs/web-platform/launch-handler/index.md:93]: Tiny nit: I decided to call this `focus-matching-url` in the explainer.

Comment (tomayac) [reply site/en/docs/web-platform/launch-handler/index.md:93]: Will change to that then. Thanks!

Comment (tomayac) [reply site/en/docs/web-platform/launch-handler/index.md:93]: See https://github.com/GoogleChrome/developer.chrome.com/pull/3378.

**Pull request merged**